spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/context/ContextCredentialsAutoConfiguration.java

```diff
@@ -85,6 +94,25 @@ public AWSCredentialsProvider awsCredentialsProvider(
 					: new ProfileCredentialsProvider(properties.getProfileName()));
 		}
 
+		if (StringUtils.hasText(properties.getRoleArn())
+				&& StringUtils.hasText(properties.getRoleSessionName())) {
+
+			AWSSecurityTokenServiceClientBuilder stsClientBuilder = AWSSecurityTokenServiceClientBuilder
+					.standard().withCredentials(buildCredentialsProviderChain(providers));
+			RegionProvider regionProvider = regionObjectProvider.getIfAvailable();
+			if (regionProvider != null) {
+				stsClientBuilder.withRegion(regionProvider.getRegion().getName());
+			}
+
+			AWSCredentialsProvider provider = new STSAssumeRoleSessionCredentialsProvider.Builder(
+					properties.getRoleArn(), properties.getRoleSessionName())
+							.withRoleSessionDurationSeconds(
+									properties.getRoleSessionDurationSeconds())
+							.withStsClient(stsClientBuilder.build()).build();
+
+			providers.add(0, provider);
```

Comment:
This provider configuration could be moved to the top of the method, then it would be easier to follow that it has a priority over other providers. Now I find it easy to miss that it adds it at index `0`.

Reply:
Yes, I agree that it would be easier, but `STSAssumeRoleSessionCredentialsProvider` depends on the providers defined above. Besides, if temporary credentials are successfully created, other providers in the `AWSCredentialsProviderChain` will be ignored. It might make sense to overwrite other providers if authorization via STS is supposed. WDYT?

Reply:
Thanks for explanation, I understand now.